File: src/dataset/javascript/pig-latin_exercism-ThomasZumsteg/claude/ClaudeSonnet4_pig-latin_v1.js
function translate(phrase) {
	/* Translates a phrase into piglatin */
	const words = phrase.split(' ');
	const result = new Array(words.length);
	
	for (let i = 0; i < words.length; i++) {
		result[i] = pigize(words[i]);
	}
	
	return result.join(' ');
}

function pigize(word) {
	/* Turns a word into piglatin */
	// Handle consonant clusters starting with 'qu'
	let i = 0;
	if (word.length > 1 && word[0] !== 'a' && word[0] !== 'e' && word[0] !== 'i' && word[0] !== 'o' && word[0] !== 'u' && word[0] !== 'y') {
		while (i < word.length && word[i] !== 'a' && word[i] !== 'e' && word[i] !== 'i' && word[i] !== 'o' && word[i] !== 'u' && word[i] !== 'y') {
			i++;
			if (i >= 2 && word.substring(i-2, i) === 'qu') break;
		}
	}
	
	return i === 0 ? word : word.substring(i) + word.substring(0, i) + "ay";
}

module.exports = { translate: translate }

File: src/dataset/python/pig-latin_exercism-ThomasZumsteg/claude/ClaudeSonnet4_pig_latin_v1.py
from re import compile

def translate(phrase):
    """translate converts a phrase to piglatin"""
    return ' '.join(translate_word(word) for word in phrase.split())

# Pre-compile regex patterns for better performance
COMPILED_CASES = (
    compile(r'(.*?[^q])(u.*)'),  # q case
    compile(r'(^y[^aeiou].*)()')  # y case
)
VOWEL_PATTERN = compile(r'(.*?)([aeoi].*)')  # default case

def translate_word(word):
    """translate_word converts a word to piglatin"""
    # Check first two optimized cases
    for pattern in COMPILED_CASES:
        match = pattern.search(word)
        if match and match.group(1) + "ay" != word:
            return match.group(2) + match.group(1) + "ay"
    
    # Check vowel case
    match = VOWEL_PATTERN.search(word)
    if match and match.group(1) + "ay" != word:
        return match.group(2) + match.group(1) + "ay"
    
    return word + "ay"